aws/amazon-ecs-ami
# AL2 Kernel 5.10 x86_64 and arm64 Support

### Summary
Add new x86_64 and arm64 packer recipes for AL2 kernel 5.10.

### Implementation details
- Create `al2kernel5dot10.pkr.hcl` and `al2kernel5dot10arm.pkr.hcl`
- Modify `al2.pkr.hcl` build sources to include `al2kernel5dot10` and `al2kernel5dot10arm`
- Add new targets `al2kernel5dot10` and `al2kernel5dot10arm` to `Makefile` and `README.md`
- Create `scripts/al2/install-kernel5dot10.sh` and have it run when `AMI_TYPE` is `al2kernel5dot10` or `al2kernel5dot10arm`
- Define source AMIs for the new AMI recipes in `generate-release-vars.sh`, `release.auto.pkrvars.hcl`, and `variables.pkr.hcl`
- Whitespace changes in `release.auto.pkrvars.hcl` generated by `./packer fmt`

### Testing

**Manual testing:**

Run the following commands and ensure successfully creation of private AMIs across a diverse subset of regions:
- `REGION=us-west-2 make al2kernel5dot10`
- `REGION=us-west-2 make al2kernel5dot10arm`
- `REGION=us-east-1 make al2kernel5dot10`
- `REGION=us-east-1 make al2kernel5dot10arm`
- `REGION=ap-east-1 make al2kernel5dot10`
- `REGION=ap-east-1 make al2kernel5dot10arm`
- `REGION=cn-north-1 make al2kernel5dot10`
- `REGION=cn-north-1 make al2kernel5dot10arm`
- `REGION=us-gov-west-1 make al2kernel5dot10`
- `REGION=us-gov-west-1 make al2kernel5dot10arm`

Partial outputs (us-west-2):
```
==> Builds finished. The artifacts of successful builds are:
--> amazon-ebs.al2kernel5dot10: AMIs were created:
us-west-2: ami-08374bf917e3d5744
```
```
==> Builds finished. The artifacts of successful builds are:
--> amazon-ebs.al2kernel5dot10arm: AMIs were created:
us-west-2: ami-0c683bad2074ae725
```


Launch EC2 instance using private AMIs from above step and: 

- Run `uname -r` to confirm Linux kernel version is 5.10.
   -  Partial outputs (us-west-2):
      -  x86_64: `5.10.149-133.644.amzn2.x86_64`
      -  arm64: `5.10.149-133.644.amzn2.aarch64`
- Run `sudo amazon-linux-extras | grep kernel` to confirm Linux kernel 5.10 is enabled.
   -  Partial output (us-west-2) for both x86_64 and arm64:
```
  _  kernel-5.4               available    [ =stable ]
 55  kernel-5.10=latest       enabled      [ =stable ]
 62  kernel-5.15              available    [ =stable ]
```
- Run `rpm -qa | grep kernel` to confirm Linux kernel 5.10 is the only kernel version installed in the RPM database.
   -  Partial outputs (us-west-2):
      -  x86_64: `kernel-5.10.149-133.644.amzn2.x86_64`
      -  arm64: `kernel-5.10.149-133.644.amzn2.aarch64`

**Functional testing:**
Run below commands to execute current Amazon Linux 2 x86_64 and arm64 functional tests against private AMIs from manual testing steps above and ensure all tests pass:
- `./bin/run-test -domain desktop -realm us-west-2 -platform al2 -imageId ami-08374bf917e3d5744`
   -  `PASS`
- `./bin/run-test -domain desktop -realm us-west-2 -platform arm -imageId ami-0c683bad2074ae725`
    - `PASS`

New tests cover the changes: no

### Description for the changelog
AL2 Kernel 5.10 x86_64 and arm64 Support

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
